src/Nest/CommonAbstractions/ConnectionSettings/PocoMapping.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace Nest
+{
+	public interface IPocoMapping
+	{
+		Type ClrType { get; }
+
+		/// <summary>
+		/// When specified dictates the default Elasticsearch index name for <see cref="ClrType"/>
+		/// </summary>
+		string IndexName { get; set; }
+
+		/// <summary>
+		/// When specified dictates the default Elasticsearch type name for <see cref="ClrType"/>
+		/// </summary>
+		string TypeName { get; set; }
+
+		/// <summary>
+		/// When specified dictates the relation name for <see cref="ClrType"/> to resolve to.
+		/// </summary>
+		string RelationName { get; set; }
+
+	}
+
+	public interface IPocoMapping<T> : IPocoMapping where T : class
+	{
+		/// <summary>
+		/// Allows you to set a default Id property on <typeparamref name="T" /> that NEST will evaluate
+		/// </summary>
+		Expression<Func<T, object>> IdProperty { get; set; }
+
+		/// <summary>
+		/// When specified allows you to ignore or rename certain properties of clr type <typeparamref name="T" />
+		/// </summary>
+		IList<IPocoPropertyMapping<T>> Properties { get; set; }
+	}
+
+	public class PocoMapping : IPocoMapping
+	{
+
+		public Type ClrType { get; }
+		public PocoMapping(Type type) => ClrType = type;
+
+		/// <summary>
+		/// When specified dictates the default Elasticsearch index name for <see cref="ClrType"/>
+		/// </summary>
+		public string IndexName { get; set; }
+
+		/// <summary>
+		/// When specified dictates the default Elasticsearch type name for <see cref="ClrType"/>
+		/// </summary>
+		public string TypeName { get; set; }
+
+		/// <summary>
+		/// When specified dictates the relation name for <see cref="ClrType"/> to resolve to.
+		/// </summary>
+		public string RelationName { get; set; }
+
+	}
+	public class PocoMapping<T> : PocoMapping, IPocoMapping<T> where T : class
+	{
+		public PocoMapping() : base(typeof(T)) { }
+
+		/// <summary>
+		/// Allows you to set a default Id property on <typeparamref name="T" /> that NEST will evaluate
+		/// </summary>
+		public Expression<Func<T, object>> IdProperty { get; set; }
+
+		/// <summary>
+		/// When specified allows you to ignore or rename certain properties of clr type <typeparamref name="T" />
+		/// </summary>
+		public IList<IPocoPropertyMapping<T>> Properties { get; set; }
```

Comment:
Should the non-generic `PocoMapping` also expose a `Properties` collection, since ultimately the properties are added to a `FluentDictionary<MemberInfo, IPropertyMapping>`?